### DISCUSSION TITLE
Faster followers/following

### DISCUSSION THREAD

**Discussion by anilsari** [sbertix/Swiftagram#147] [Q&A] [answered]
Hi Sbertix
I have a problem.When i try to get user's followers ( she has 4K followers ) i am waiting almost 60-65 seconds.How can i make it faster? **This is my code :** 

```
Endpoint.Friendship.following(secret.identifier)
            .unlocking(with: secret)
            .task(maxLength: .max,
                  onComplete: { _ in
                    
                    let methodFinish = Date()
                    let executionTime = methodFinish.timeIntervalSince(methodStart)
                    print("Execution time: \(executionTime)")
                    
                    let alert = UIAlertController(title: "Alert", message: "Execution time: \(executionTime)", preferredStyle: .alert)
                    self.present(alert, animated: true, completion: nil)
                  },
                  onChange: { [weak self] in     
                    let data = try? $0.get()
                    data?.users?.forEach({user in
                        print(user.identifier, user.username)
                    })
            })
            .resume()
```

**Comment by sbertix** [comment, marked as answer]
Hey @anilsari,

Assuming you're not actually `print`ing the results in the your code (as printing each element to the console is of course a lot slower than processing it), it's a valid question. 

**Swiftagram** is based on Instagram Private APIs, which are used by the official apps to deal with user interaction. They're not meant for bots and they're severely limited to prevent abuses. 
So, unfortunately there's not a quicker way to deal with it using them – especially for friendship endpoints, which are heavily rate limited. 

Most apps relying on "quick" followers/following pagination tend to use a scraper instead. The implementation is pretty straightforward, if you base it on Instagram mobile website, but it's beyond the scopes of this project.
